includes/Stories_Renderer/Renderer.php

```diff
@@ -0,0 +1,368 @@
+	 *
+	 * @return array Returns single story item data.
+	 */
+	protected function get_story_item_data( $story_id, $single_story_classes = '' ) {
+
+		$story_data = [];
+
+		if ( empty( $story_id ) ) {
+			return $story_data;
+		}
+
+		$author_id       = absint( get_post_field( 'post_author', $story_id ) );
+		$is_circles_view = $this->is_view_type( 'circles' );
+		$image_size      = $is_circles_view ? Media::POSTER_SQUARE_IMAGE_SIZE : Media::POSTER_PORTRAIT_IMAGE_SIZE;
+		$story_title     = ( ! empty( $this->attributes['show_title'] ) && ( true === $this->attributes['show_title'] ) ) ?
+			get_the_title( $story_id ) :
+			'';
+		$author_name     = ( ! $is_circles_view && ! empty( $this->attributes['show_author'] ) && ( true === $this->attributes['show_author'] ) ) ?
+			get_the_author_meta( 'display_name', $author_id ) :
+			'';
+		$story_date      = ( ! $is_circles_view && ! empty( $this->attributes['show_date'] ) && ( true === $this->attributes['show_date'] ) ) ?
+			get_the_date( 'M j, Y', $story_id ) :
+			'';
+
+		$story_data['ID']                   = $story_id;
+		$story_data['url']                  = get_post_permalink( $story_id );
+		$story_data['title']                = $story_title;
+		$story_data['height']               = '430';
+		$story_data['width']                = '285';
+		$story_data['poster']               = get_the_post_thumbnail_url( $story_id, $image_size );
+		$story_data['author']               = $author_name;
+		$story_data['date']                 = $story_date;
+		$story_data['class']                = $single_story_classes;
+		$story_data['show_content_overlay'] = ( ! empty( $story_title ) || ! empty( $author_name ) || ! empty( $story_date ) );
+
+		return $story_data;
+	}
+
+	/**
+	 * Verifies the current view type.
+	 *
+	 * @param string $view_type View type to check.
+	 *
+	 * @return bool Whether or not current view type matches the one passed.
+	 */
+	protected function is_view_type( $view_type ) {
+		return ( ! empty( $this->attributes['view_type'] ) && $view_type === $this->attributes['view_type'] );
+	}
+
+	/**
+	 * Get view type for stories.
+	 *
+	 * @return string
+	 */
+	protected function get_view_type() {
+		return isset( $this->attributes['view_type'] ) ? $this->attributes['view_type'] : '';
+	}
+
+	/**
+	 * Renders stories archive link if the 'show_stories_archive_link' attribute is set to true.
+	 *
+	 * @return void
+	 */
+	protected function maybe_render_archive_link() {
+		if ( ( ! empty( $this->attributes['show_stories_archive_link'] ) ) && ( true === $this->attributes['show_stories_archive_link'] ) ) :
+			$web_stories_archive = get_post_type_archive_link( Story_Post_Type::POST_TYPE_SLUG );
+			$web_stories_archive = ( is_string( $web_stories_archive ) ? $web_stories_archive : '' );
+			?>
+			<div class="web-stories__archive-link">
+				<a href="<?php echo( esc_url_raw( $web_stories_archive ) ); ?>">
+					<?php echo( esc_html( $this->attributes['stories_archive_label'] ) ); ?>
+				</a>
```

Comment:
If `$web_stories_archive` is an empty string (`''`), we'll end up with `<a href="">`.

Let's not print any link if there is none.